### ISSUE TITLE
Change event that doesn't conform to type

### ISSUE THREAD

**Issue by erezarnon** [janosh/svelte-multiselect#276]
The component forwards change events from its input. Those events don't conform to the change event type definition.

## Repro

Steps to reproduce the problem:

1. Add a change handler that uses `e.detail.type`, which according the documentation should exist
2. Click on the multiselect's input field
3. Type in some characters
4. Blur
5. The input's change event gets emitted. `e.detail` is undefined and you get an error.

**Comment by janosh** [comment]
You're confusing two types of `change` events. The component emits synthetic `change` events whenever the `selected` options list changes (i.e. items are added/removed). This one has a `event.detail.type` payload. It also forwards DOM `change` events from the `input` node. This one _does not_ have a `event.detail.type` payload.

Maybe the docs could be clearer on this point. PRs welcome!

**Comment by erezarnon** [comment]
Yup, we had coded based on the docs and every once in a while would get exceptions that didn't make sense. I had to look at the source code to figure out how `detail` can be `undefined`, and once that was clear it was easy to reproduce and handle the input change events differently.

**Comment by erezarnon** [comment]
Actually I think the a better solution would be to rename the synthetic change event so each event has one consistent type instead of putting 2 separate things in 1 interface
